--- a/deps/linenoise/Makefile
+++ b/deps/linenoise/Makefile
@@ -6,7 +6,7 @@ R_CFLAGS= $(STD) $(WARN) $(OPT) $(DEBUG)
 R_LDFLAGS= $(LDFLAGS)
 DEBUG= -g
 
-R_CC=$(CC) $(R_CFLAGS)
+R_CC=$(CC) $(R_CFLAGS) $(CPPFLAGS)
 R_LD=$(CC) $(R_LDFLAGS)
 
 linenoise.o: linenoise.h linenoise.c
--- a/src/Makefile
+++ b/src/Makefile
@@ -263,7 +263,7 @@ else
     endef
 endif
 
-REDIS_CC=$(QUIET_CC)$(CC) $(FINAL_CFLAGS)
+REDIS_CC=$(QUIET_CC)$(CC) $(FINAL_CFLAGS) $(CPPFLAGS)
 REDIS_LD=$(QUIET_LINK)$(CC) $(FINAL_LDFLAGS)
 REDIS_INSTALL=$(QUIET_INSTALL)$(INSTALL)
 
